internal/private/private.go

```diff
@@ -112,10 +112,10 @@ type TryReusingBlobOptions struct {
 	// Transports, OTOH, MUST support these fields being zero-valued for types.ImageDestination callers
 	// if they use internal/imagedestination/impl.Compat;
 	// in that case, they will all be consistently zero-valued.
```

Comment:
A compatibility handler for external `ImageDestinations` is missing for the new field:

`internal/imagedestination/wrapper.go`. I think if `RequiredCompression` is set, the call should silently return without reusing anything; we could theoretically call `TryReusingBlob` and checking whether the returned blob has the right algorithm, but that’s AFAICS unsafe — `ImageDestination` is stateful and the destination can assume, that if `TryReusingBlob` returns something, it is going to be used. (Compare `storageImageDestination`.)

Reply:
Added compat handler